src/libraries/System.Net.Http/src/System/Net/Http/HttpMetrics.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpMetrics
+    {
+        private readonly Meter _meter;
+        private readonly UpDownCounter<long> _currentRequests;
+        private readonly Histogram<double> _requestsDuration;
+
+        public HttpMetrics(Meter meter)
+        {
+            _meter = meter;
+
+            _currentRequests = _meter.CreateUpDownCounter<long>(
+                "current-requests",
+                description: "Number of outbound HTTP requests that are currently active on the client.");
+
+            _requestsDuration = _meter.CreateHistogram<double>(
+                "request-duration",
+                unit: "s",
+                description: "The duration of outbound HTTP requests.");
+        }
+
+        public void RequestStart(HttpRequestMessage request)
+        {
+#pragma warning disable SA1129 // Do not use default value type constructor
+            var tags = new TagList();
+#pragma warning restore SA1129 // Do not use default value type constructor
+            InitializeCommonTags(ref tags, request);
+            _currentRequests.Add(1, tags);
+        }
+
+        public void RequestStop(HttpRequestMessage request, HttpResponseMessage? response, Exception? unhandledException, long startTimestamp, long currentTimestamp)
+        {
+#pragma warning disable SA1129 // Do not use default value type constructor
+            var tags = new TagList();
+#pragma warning restore SA1129 // Do not use default value type constructor
+            InitializeCommonTags(ref tags, request);
+            _currentRequests.Add(-1, tags);
+
+            if (response is not null)
+            {
+                tags.Add("status-code", (int)response.StatusCode); // Boxing?
+                tags.Add("protocol", $"HTTP/{response.Version}"); // Hacky
```

Comment:
Hacky because it relies on formatting implementation? Shall we do something like this to avoid the formatting allocation?

```C#
static string GetProtocolName6(Version httpVersion) => (httpVersion.Major, httpVersion.Minor) switch
{
    (1, 1) => "HTTP/1.1",
    (2, 0) => "HTTP/2.0",
    (3, 0) => "HTTP/3.0",
    _ => "unknown"
};
```

Reply:
Hacky because it allocates some values aren't right. `HTTP/2` would be better than `HTTP/2.0`.